Development/cpprest/basic_utils.h

```diff
@@ -30,6 +30,43 @@ namespace utility
                 return !iss.fail() ? t : default_val;
             }
         }
+
+        /// <summary>
+        /// Encode the given byte array into a base64url string
+        /// using the alternative alphabet and skipping the padding
+        /// as per https://tools.ietf.org/html/rfc4648#section-5
+        /// </summary>
+        inline utility::string_t to_base64url(const std::vector<unsigned char>& data)
+        {
+            auto str = utility::conversions::to_base64(data);
+            auto it = str.begin();
+            for (; str.end() != it; ++it)
+            {
+                auto& c = *it;
+                if (U('=') == c) break;
+                if (U('+') == c) c = U('-');
+                else if (U('/') == c) c = U('_');
+            }
+            str.erase(it, str.end());
+            return str;
+        }
+
+        /// <summary>
+        /// Decode the given base64url string to a byte array
+        /// using the alternative alphabet and skipping the padding
+        /// as per https://tools.ietf.org/html/rfc4648#section-5
+        /// </summary>
```

Comment:
Compile with /doc to process documentation comments to a file. See https://learn.microsoft.com/en-us/cpp/build/reference/summary-visual-cpp?view=msvc-170

We could remove the `<summary>` tag and triple-slash because we are not generating documents.